common/components/BalanceSidebar/BalanceSidebar.jsx

```diff
@@ -28,12 +28,20 @@ type Props = {
 export class BalanceSidebar extends React.Component {
   props: Props;
   state = {
-    showLongBalance: false
+    showLongBalance: false,
+    address: ''
   };
 
+  componentDidMount() {
+    this.props.wallet.getAddress().then(address => {
```

Comment:
Anytime we display an address it needs to be checksumm'd address

Anytime we interact with an address (e.g. hashes, blockies) it needs to be all lowercase address.

https://github.com/ethereumjs/ethereumjs-util/blob/cfeac0fc49c00111a77124bbe76011e4b1129cf9/index.js#L441-L455

When a user types in an address, it should display error if not checksummed and different error if completely invalid.

> This address is not checksummed. If you hand-typed the address, please go back and copy & paste it. Your funds are not recoverable if you send to an invalid or wrong address.

> This address is not valid. The address should look similar to: 0x7cB57B5A97eAbe94205C07890BE4c1aD31E486A8